Makefile

```diff
@@ -427,20 +427,22 @@ __check_defined = \
 	$(if $(value $1),, \
 		$(error Undefined environment variable $1$(if $2, ($2))))
 
-generate-hypershift-deplyment:
-	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
-	@:$(call check_defined, TMPDIR)
+# yq: https://github.com/mikefarah/yq
+yq ?= yq
 
-	mkdir -p ${TMPDIR}/hypershift-cli
-	oc image extract registry.ci.openshift.org/ci/hypershift-cli:latest --path /usr/bin/hypershift:${TMPDIR}/hypershift-cli --confirm
-	chmod +x ${TMPDIR}/hypershift-cli/hypershift
+generate-hypershift-deployment:
+	@:$(call check_defined, MGMT_AWS_CONFIG_PATH)
```

Comment:
I knew this is not introduced in this PR, but it would be much simpler if `MGMT_AWS_CONFIG_PATH` is defined with a default value (you could use the one when you call it from your laptop as you are probably the one most often) as we do for many of others. Thus, we do not need `check_defined` and `__check_defined` function at all.

I also checked the linked stackoverflow page. Even the following as the other suggestion looks more reasonable to me:
- I do not need to jump to another part of the makefile to understand what it does.
- The two `check` functions themselves are too esoteric. If there is a simple way of avoiding them, I would go there.

```
ifndef MY_FLAG
$(error MY_FLAG is not set)
endif
```

Reply:
`MGMT_AWS_CONFIG_PATH` has no and should have no default value. That's why its `mandatory`.

> I do not need to jump to another part of the makefile to understand what it does.
You don't need to. The name of that function already suggests the purpose of it

> The two check functions themselves are too esoteric. If there is a simple way of avoiding them, I would go there
I don't see repeatedly calling `ifndef` is better than using a function